examples/with-next/pages/index.tsx

```diff
@@ -193,69 +190,61 @@ export default class Index extends PureComponent {
   startKeyword = async () => {
     this.initialize()
     this.setState({ keyword: { status: 'Calibrating...' } })
-    const speechConfig: SpeechConfig = DefaultSpeechConfig
-    speechConfig.baseKeywordUrl = 'https://s.spokestack.io/u/UbMeX/js'
-    speechConfig.keywordClasses = [
-      'zero',
-      'one',
-      'two',
-      'three',
-      'four',
-      'five',
-      'six',
-      'seven',
-      'eight',
-      'nine'
-    ]
-    const config: PipelineConfig = {
-      onReady: () => {
-        this.setState({ keyword: { status: 'Listening', result: '' } })
-      },
-      onEvent: (evt) => {
-        const { eventType, transcript } = evt
-        if (eventType == EventType.Recognize) {
-          this.setState({ keyword: { error: '', result: transcript } })
-        } else if (eventType == EventType.Timeout) {
-          this.setState({ keyword: { error: 'timeout' } })
-        }
-      },
-      speechConfig: speechConfig,
-      stages: [Stage.VadTrigger, Stage.KeywordRecognizer]
+    const profile = PipelineProfile.Keyword
+    profile.speechConfig = {
+      ...profile.speechConfig,
+      baseKeywordUrl: 'https://s.spokestack.io/u/UbMeX/js',
+      keywordClasses: [
+        'zero',
+        'one',
+        'two',
+        'three',
+        'four',
+        'five',
+        'six',
+        'seven',
+        'eight',
+        'nine'
+      ]
+    }
+    profile.onReady = () => {
+      this.setState({ keyword: { status: 'Listening', result: '' } })
     }
-    this.pipeline = new Pipeline(config)
-    this.pipeline.listen()
+    profile.onEvent = (evt) => {
+      const { eventType, transcript } = evt
+      if (eventType == EventType.Recognize) {
+        this.setState({ keyword: { error: '', result: transcript } })
+      } else if (eventType == EventType.Timeout) {
+        this.setState({ keyword: { error: 'timeout' } })
+      }
+    }
+
+    startPipeline(profile)
```

Comment:
Keyword would look like this...

```ts
await startPipeline({
  profile: PipelineProfile.Keyword,
  baseUrl: 'https://s.spokestack.io/u/UbMeX/js',
  keywords: [
    'zero',
    'one',
    'two',
    'three',
    'four',
    'five',
    'six',
    'seven',
    'eight',
    'nine'
  ],
  onEvent: (evt) => { ... }
})
this.setState({ keyword: { status: 'Listening', result: '' } })
```

Note: the promise should be rejected if the pipeline fails to start.

Now that I think about it, if there will never be more than 2 profiles, we could simplify this further and make it...
```ts
await startPipeline({
  isWakeword: false,
  // ...
})
```

Reply:
1. How would you customize portions of the speech config if you did it that way?
2. There should eventually be many more than 2 profiles (see the [Android version](https://github.com/spokestack/spokestack-android/tree/master/src/main/java/io/spokestack/spokestack/profile)). I'd like to add ASR stages with as many different providers as possible—this is just a baby pipeline, but it seemed reasonable to introduce the structure now to reduce refactoring later.

Reply:
1. Speech config properties can be passed as options.
2. Fair enough, let's keep profile then, but I also wonder if there is a sensible default so that one of these cases does not need to pass `profile`.

Reply:
Another concern is that `baseWakewordUrl` and `baseKeywordUrl` are being collapsed—you could conceivably use them together, though that's not supported by a profile atm (another reason I'd like to keep exposing the low-level components, so people can put together their own stage lists).

Reply:
You can accomplish #1 in a few ways, but the first that comes to mind would allow you to use all *remaining* options as speech config properties. For example,

```ts
function startPipeline({ profile, baseUrl, onEvent, ...rest }) {
  const speechConfig = {
    ...defaultSpeechConfig,
    ...rest
  }
}
```

Reply:
I'm also fine if we want to add `speechConfig` as an option itself, but I'd prefer to keep commonly passed options separate and put them on `speechConfig` if necessary ourselves.

```ts
function startPipeline({ profile, speechConfig, etc }) {
  speechConfig = {
    ...defaultSpeechConfig,
    profile
  }
}
```

Reply:
Now that I look at the example again, the second option is probably better.

Reply:
And it's `baseUrl` that would get assigned to `speechConfig` by us.

Reply:
My comments here are resolved. New comments take their place.